core/src/main/scala/org/apache/spark/crypto/CryptoConf.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.security.NoSuchAlgorithmException
+import javax.crypto.KeyGenerator
+
+import org.apache.hadoop.security.Credentials
+
+import org.apache.spark.SparkConf
+import org.apache.spark.crypto.CommonConfigurationKeys._
+
+/**
+ * CryptoConf is a class for Crypto configuration
+ */
+private[spark] object CryptoConf {
+  def isShuffleEncryted(sparkConf: SparkConf): Boolean = {
+    if (sparkConf != null) {
+      sparkConf.getBoolean(SPARK_ENCRYPTED_SHUFFLE, false)
+    } else {
+      false
+    }
+  }
+
+  def initSparkShuffleCredentials(conf: SparkConf, credentials: Credentials) {
```

Comment:
I see you're calling this method both in `Client` and in `ApplicationMaster`. Is that necessary? Jobs will always go through `Client`, nobody runs the `ApplicationMaster` directly.

If the two calls are not necessary, you could remove the check for whether the key has already been added to the credentials object.